app/src/main/java/nu/yona/app/ui/profile/EditDetailsProfileFragment.java

```diff
@@ -138,107 +144,65 @@ public void afterTextChanged(Editable s)
 				}
 			}
 		};
-
-		onFocusChangeListener = new View.OnFocusChangeListener()
-		{
-			@Override
-			public void onFocusChange(View view, boolean b)
-			{
-				if (b)
-				{
-					((EditText) view).setSelection(((EditText) view).getText().length());
-				}
-			}
-		};
-		inflateView(view);
-
-		setHook(new YonaAnalytics.BackHook(AnalyticsConstant.BACK_FROM_EDIT_PROFILE));
-
-		YonaApplication.getEventChangeManager().registerListener(this);
-		return view;
 	}
 
 	private void inflateView(View view)
 	{
+		setupFirstNameLayout(view);
+		setupLastNameLayout(view);
+		setupNickNameLayout(view);
+		setupMobileNumberLayout(view);
+		profileImage = view.findViewById(R.id.profileImage);
+		updateProfileImage = view.findViewById(R.id.updateProfileImage);
+		profileImageTxt = view.findViewById(R.id.profileIcon);
+		profileEditMode();
+	}
 
-		firstnameLayout = view.findViewById(R.id.first_name_layout);
-		lastNameLayout = view.findViewById(R.id.last_name_layout);
-		nickNameLayout = view.findViewById(R.id.nick_name_layout);
-		mobileNumberLayout = view.findViewById(R.id.mobile_number_layout);
-
+	private void setupFirstNameLayout(View view)
+	{
+		firstNameLayout = view.findViewById(R.id.first_name_layout);
 		firstName = view.findViewById(R.id.first_name);
 		firstName.addTextChangedListener(textWatcher);
-		firstName.setOnFocusChangeListener(onFocusChangeListener);
+		firstName.setOnFocusChangeListener(EditDetailsProfileFragment::onFocusChange);
```

Comment:
By applying this pattern everywhere, you don't need the ``onFocusChangeListener`` member anymore.